src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/DefaultRazorSemanticTokenInfoService.cs

```diff
@@ -0,0 +1,197 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Language;
+using Microsoft.AspNetCore.Razor.Language.Legacy;
+using Microsoft.AspNetCore.Razor.Language.Syntax;
+using Microsoft.AspNetCore.Razor.LanguageServer.Completion;
+using Microsoft.VisualStudio.Editor.Razor;
+using SyntaxNode = Microsoft.AspNetCore.Razor.Language.Syntax.SyntaxNode;
+
+namespace Microsoft.AspNetCore.Razor.LanguageServer.Semantic
+{
+    internal class DefaultRazorSemanticTokenInfoService : RazorSemanticTokenInfoService
+    {
+        private readonly SemanticTokenLegend _symanticTokenLegend;
+
+        [ImportingConstructor]
```

Comment:
Why this attribute?

Reply:
It indicates which constructor to use when creating an object for DI. It's true that [it would be the default](https://docs.microsoft.com/en-us/dotnet/api/system.composition.importingconstructorattribute?view=dotnet-plat-ext-3.1#remarks) as a parameter-less constructor, but I like to be explicit with this kind of thing.

Reply:
I don't believe our language server pieces respect the `[ImportingConstructor]` piece given their DI is based on ASP.NET Core's dependency injection system and not MEF

Reply:
But we don't use the Composition model of service registration here right? (We don't `[Export]` this contract). From my understanding, we use `ServiceCollection` which handles DI differently and `[ImportingConstructor]` attribute will have no effect here.